### PR TITLE
[#14] Support Hound

### LINKED ISSUE #14: Add hound integration test configuration option
We should support an option that would configure integration testing through hound if you are generating a web domain. Headless chrome has made integration testing far less painful.

I'm still getting a handle on how mithril is setup, so I'm not quite clear how/where web integration testing should be configured.

### PR DESCRIPTION
**Pull request by danielberkompas**
Adds a `--integration hound` generator option which will:

- Add a check to the `bin/setup` script for `webdriver`
- Configure Hound to use Chrome WebDriver (the only reliable driver
we've found, in our experience)
- Create an `IntegrationCase`
- Create a sample test on the home page
- Update `bin/test` to automatically start/stop WebDriver

Closes #14.

### CODE REVIEW

File: template/$PROJECT_NAME$/apps/$PROJECT_NAME$_web/test/support/integration_case.ex [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  use ExUnit.CaseTemplate
+
+  using do
+    quote do
+      use <%= @project_name_camel_case %>Web.ConnCase
+      use Hound.Helpers
+
+      @moduletag :integration
+
+      hound_session()
```
Comment (zberkom) [inline template/$PROJECT_NAME$/apps/$PROJECT_NAME$_web/test/support/integration_case.ex]: We may want to leave a comment describing the configuration necessary to run the session headlessly. 

```
additional_capabilities: %{
          chromeOptions: %{ "args" => [
            "--user-agent=#{Hound.Browser.user_agent(:chrome)}",
            "--headless",
            "--disable-gpu"
            ]
          }
        }
```

Comment (danielberkompas) [reply template/$PROJECT_NAME$/apps/$PROJECT_NAME$_web/test/support/integration_case.ex]: According to the docs, it looks like this is what you need to do:

```elixir
# Use Chrome in headless mode with ChromeDriver (default port 9515 assumed) 
config :hound, driver: "chrome_driver", browser: "chrome_headless"
```

Comment (zberkom) [reply template/$PROJECT_NAME$/apps/$PROJECT_NAME$_web/test/support/integration_case.ex]: Ah, they finally merged that update then. It wasn't possible to do that when I first implemented it.

File: template/$PROJECT_NAME$/bin/test [hunk 2]
```diff
@@ -3,10 +3,16 @@
 # Test Script
 #
 # Runs all commands needed to fully test the application.
+<%= if assigns[:integration] == "hound" do %>
+# Automatically start Chrome Driver in the background
+chromedriver &
+
+# Automatically kill Chrome Driver after script exits
+trap 'jobs -p | xargs kill' EXIT
```
Comment (zberkom) [inline template/$PROJECT_NAME$/bin/test]: Nice! 🎉

**Pull request merged**